src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Repeater/Given_ItemsRepeater.cs

```diff
@@ -114,7 +115,123 @@ public async Task When_NestedInSVAndOutOfViewportOnInitialLoad_Then_Materialized
+									Child = new TextBlock().Apply(tb => tb.SetBinding(TextBlock.TextProperty, new Binding()))
+								})
+							}
+						}
+					})
+				})
+			};
+
+			TestServices.WindowHelper.WindowContent = sv;
+			await TestServices.WindowHelper.WaitForIdle();
+
+			sv.ChangeView(null, sv.ExtentHeight/2, null, disableAnimation: true);
+			await TestServices.WindowHelper.WaitForIdle();
+
+			var groupView = sut.Children.Single(g => g.DataContext as string == "Group #05");
+			var groupIr = (ItemsRepeater)((StackPanel)groupView).Children[1];
+
+			var beforeVisibleItems = groupIr.Children.Select(i => i.DataContext?.ToString()).Order().ToArray();
+
+			// Scroll by baby step to not be above the threshold which would cause a complete redraw
+			const int step = 10;
+			for (var i = 0; i < viewportHeight * 5; i += step)
+			{
+				sv.ChangeView(null, sv.VerticalOffset + step, null, disableAnimation: true);
+				await TestServices.WindowHelper.WaitForIdle();
+			}
+
+			var afterVisibleItems = groupIr.Children.Select(i => i.DataContext?.ToString()).Order().ToArray();
+
+			afterVisibleItems.Should().NotContain(beforeVisibleItems);
+		}
+#endif
+
+
+		[TestMethod]
+		[RunsOnUIThread]
+#if !HAS_UNO
+		[Ignore("Custom behavior of uno")]
+#elif __MACOS__
+		[Ignore("Currently fails on macOS, part of #9282 epic")]
+#endif
+		public async Task When_UnloadAndReload_Then_UnsubscribeAndResubscribeToEffectiveViewportChanged()
+		{
+			var evt = typeof(FrameworkElement).GetField("_effectiveViewportChanged", BindingFlags.Instance | BindingFlags.NonPublic)
+				?? throw new InvalidOperationException("Cannot find the private event backing field.");
+
+			var sut = default(ItemsRepeater);
+			var root = new Border
+			{
+				Child = (sut = new ItemsRepeater
+				{
+					ItemsSource = Enumerable.Range(0, 10).Select(i => $"Item #{i}"),
+					Layout = new StackLayout { Orientation = Orientation.Horizontal },
+					ItemTemplate = new DataTemplate(() => new Border
+					{
+						Width = 100,
+						Height = 100,
+						Background = new SolidColorBrush(Colors.DeepSkyBlue),
+						Margin = new Thickness(10),
+						Child = new TextBlock().Apply(tb => tb.SetBinding(TextBlock.TextProperty, new Binding()))
+					})
+				})
+			};
+
+			TestServices.WindowHelper.WindowContent = root;
+			await TestServices.WindowHelper.WaitForIdle();
+
+			evt.GetValue(sut).Should().NotBeNull();
+
+			// Unload the IR
+			root.Child = new TextBlock { Text = "IR unloaded" };
+			await TestServices.WindowHelper.WaitForIdle();
+
+			evt.GetValue(sut).Should().BeNull("the ViewportManagerWithPlatformFeatures should have remove handler in the ResetScrollers method");
+
+			// Load again IR
+			root.Child = sut;
+			await TestServices.WindowHelper.WaitForIdle();
+
+			evt.GetValue(sut).Should().NotBeNull("the IR should have invalidate it's measure, causing a layout pass driving to invoke the ViewportManagerWithPlatformFeatures.EnsureScroller which should have re-added handler");
```

Comment:
```suggestion
			evt.GetValue(sut).Should().NotBeNull("The IR should have invalidated its measure, causing a layout pass driving to invoke the ViewportManagerWithPlatformFeatures.EnsureScroller which should have re-added handler");
```

Reply:
@agneszitte we should not upper case the first letter for the `reason` in fluent assertion: they are output into a full sentence. But reading the doc, it appears that we should also start the sentence by "because" ...

![image](https://user-images.githubusercontent.com/8635919/236467166-99744cb9-3f89-41d3-a5d1-9c5710a68961.png)

https://fluentassertions.com/introduction

```suggestion
			evt.GetValue(sut).Should().NotBeNull("because the IR should have invalidated its measure, causing a layout pass driving to invoke the ViewportManagerWithPlatformFeatures.EnsureScroller which should have re-added handler");
```